crypto/keys/keys_test.go

```diff
@@ -324,3 +324,25 @@ func TestGenerateKey(t *testing.T) {
 		}
 	}
 }
+
+func TestMarshalPrivateKey(t *testing.T) {
+	for _, tc := range []struct {
+		pem     string
+		typeURL string
+	}{
+		{ecdsaPrivateKey, "type.googleapis.com/keyspb.PrivateKey"},
+		{rsaPrivateKey, "type.googleapis.com/keyspb.PrivateKey"},
+	} {
+		k, err := NewFromPrivatePEM(tc.pem, "")
+		if err != nil {
+			t.Errorf("NewFromPrivatePEM(%v): %v", tc.pem, err)
+		}
+		a, err := MarshalPrivateKey(k)
+		if err != nil {
+			t.Errorf("MarshalPrivateKey(%v): %v", k, err)
+		}
+		if got, want := a.TypeUrl, tc.typeURL; got != want {
```

Comment:
I added a check to make sure the type contained in the DER was correct. I hope that's close enough?